registry/storage/driver/middleware/cloudfront/middleware.go

```diff
@@ -17,13 +18,15 @@ import (
 	dcontext "github.com/docker/distribution/context"
 	storagedriver "github.com/docker/distribution/registry/storage/driver"
 	storagemiddleware "github.com/docker/distribution/registry/storage/driver/middleware"
+	"github.com/hashicorp/go-version"
 )
 
 // cloudFrontStorageMiddleware provides a simple implementation of layerHandler that
 // constructs temporary signed CloudFront URLs from the storagedriver layer URL,
 // then issues HTTP Temporary Redirects to this CloudFront content URL.
 type cloudFrontStorageMiddleware struct {
 	storagedriver.StorageDriver
+	awsIPs    aWSIPs
```

Comment:
Can this be a pointer type and have it be nil when acceptedRegions is not provided? It looks like where it is used it can easily include a nil check.

Reply:
If the acceptedRegions is not provided, namely a empty string, it means requests from any regions uses S3. We would still need to check if the IP is from AWS or not.